plugins/main/tap/tap.go

```diff
@@ -0,0 +1,456 @@
+// Copyright 2022 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"syscall"
+
+	"github.com/opencontainers/selinux/go-selinux"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	current "github.com/containernetworking/cni/pkg/types/100"
+	"github.com/containernetworking/cni/pkg/version"
+	"github.com/vishvananda/netlink"
+
+	"github.com/containernetworking/plugins/pkg/ip"
+	"github.com/containernetworking/plugins/pkg/ipam"
+	"github.com/containernetworking/plugins/pkg/ns"
+	bv "github.com/containernetworking/plugins/pkg/utils/buildversion"
+	"github.com/containernetworking/plugins/pkg/utils/sysctl"
+)
+
+type NetConf struct {
+	types.NetConf
+	MultiQueue     bool    `json:"multiQueue"`
+	MTU            int     `json:"mtu"`
+	Mac            string  `json:"mac,omitempty"`
+	Owner          *uint32 `json:"owner,omitempty"`
+	Group          *uint32 `json:"group,omitempty"`
+	SelinuxContext string  `json:"selinuxContext,omitempty"`
+	Args           struct {
+		Mac string `json:"mac,omitempty"`
+	} `json:"args,omitempty"`
```

Comment:
Thank you for adding 'args'. In addition, how about to add this args under cni, as `tuning`, `host-local`, `static` and others does, to make args format consistent among this repo?

https://github.com/containernetworking/plugins/blob/c4d24e80d64393d2c632a825a3486d1c2c0248ec/plugins/ipam/static/main.go#L40-L42

Reply:
Yes, but not in this PR.

Reply:
I'm not clear what you said. As far as I see your PR, you have Args as follows:

```
	Args           struct {
		Mac string `json:"mac,omitempty"`
	} `json:"args,omitempty"`
```

I just propose it as following. Adding 'cni', to consistent args format among the plugins.

```
	Args           struct {
                A * struct {
		   Mac string `json:"mac,omitempty"`
                } `json:"cni,omitempty"`
	} `json:"args,omitempty"`
```